File: AdvancePython/Day-3 Advanced OOP concepts/Task-3-4.py
# Task 3:
# 4. Method Overriding:
class A:
    def func1(self) -> None:
        print("Feature 1 of class A")

    def func2(self) -> None:
        print("Feature 2 of class A")


class B(A):
    def func1(self) -> None:
        print("Modified feature 1 of class A by class B")

    def func3(self) -> None:
        print("Feature 3 of class B")


obj = B()
obj.func1()

File: node.js/Day-3/Task-3-4.js
// Task 3:
// 4. Method Overriding:

class A {
    func1() {
        console.log("Feature 1 of class A");
    }

    func2() {
        console.log("Feature 2 of class A");
    }
}

class B extends A {
    func1() {
        console.log("Modified feature 1 of class A by class B");
    }

    func3() {
        console.log("Feature 3 of class B");
    }
}

const obj = new B();
obj.func1();
obj.func2();